### PR TITLE
Implement Hessian matrix for polynomials

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
As requested by @wdecker .

### CODE REVIEW

File: src/Rings/mpoly.jl [hunk 1]
```diff
@@ -1294,3 +1294,20 @@ end
 
 # assure compatibility with generic code for MPolyQuos:
 lift(f::MPolyRingElem) = f
+
+### Hessian matrix
+function hessian_matrix(f::MPolyElem)
+  R = parent(f)
+  n = nvars(R)
+  df = jacobi_matrix(f)
+  result = zero(MatrixSpace(R, n, n))
```
Comment (lgoettgens) [inline src/Rings/mpoly.jl]:
```suggestion
  result = zero_matrix(R, n, n)
```
This saves the creation of the matrix space.

**Review by wdecker — APPROVED** [review]
@HechtiDerLachs thx; what about the suggestion by @lgoettgens ?

File: src/Rings/mpoly.jl (hunk 1 above)
Comment (HechtiDerLachs) [reply src/Rings/mpoly.jl]: Thanks! I was not aware of this constructor. Probably this has to do with the rather recent decision to drop parent objects for matrices over rings? Either way, this is reasonable.

Comment (thofma) [reply src/Rings/mpoly.jl]: This has been there for a few years

Comment (fingolfin) [reply src/Rings/mpoly.jl]: And which "decision to drop parent objects for matrices over rings" ? I am not aware of such a decision.

**Comment by afkafkafk13** [comment]
to be reopened

**Comment by afkafkafk13** [comment]
and reopen

**Pull request merged**